service/src/main/python/jcarbon/tensorflow/callbacks.py

```diff
@@ -125,6 +97,98 @@ def on_epoch_end(self, epoch, logs=None):
         self.reports[epoch] = self.last_report.to_frame().assign(
             epoch=epoch).set_index('epoch', append=True)
 
+class JCarbonExperimentCallback(JCarbonChunkingCallback):
+    def __init__(
+            self,
+            addr='localhost:8980',
+            period_ms=DEFAULT_PERIOD_MS,
+            signals=DEFAULT_SIGNALS,
+            chunking_period_sec=DEFAULT_PERIOD_SECS):
+        super().__init__(addr, period_ms, signals)
+        self.timestamps = {}
+        self.batch_timestamps = None
+
+    def on_epoch_begin(self, epoch, logs=None):
+        super().on_epoch_begin(epoch, logs)
+        self.curr_batch_timestamps = None
```

Comment:
What purpose does this field serve?

Reply:
I collect all the batch timestamps of one epoch in here before appending to self.batch_timestamps at on_epoch_end. Not sure how I would do it without this. Maybe the name is misleading I think